gtsam/base/ConcurrentMap.h

```diff
@@ -29,14 +29,22 @@
 #  undef max
 #  undef ERROR
 
+#include <functional> // std::hash()
+
 // Use TBB concurrent_unordered_map for ConcurrentMap
-#  define CONCURRENT_MAP_BASE tbb::concurrent_unordered_map<KEY, VALUE>
+template <typename KEY, typename VALUE>
+using CONCURRENT_MAP_BASE = tbb::concurrent_unordered_map<
```

Comment:
Naming: I think we have to use ConcurrentMapBase now that it is no longer a macro. I don’t think any projects would use the Macro, so backwards compatibility is not an issue

Reply:
Sure, I also like that name more.